server/app/middlewares/websocket_backend.rb

```diff
@@ -237,23 +237,22 @@ def subscribe_to_rpc_channel
 
   # @param [Hash] msg
   def on_rpc_message(msg)
-    client = client_for_id(msg['id'])
-    if client
-      self.send_message(client[:ws], msg['message'])
-    end
+    EM.next_tick{
+      client = client_for_id(msg['id'])
+      if client
+        self.send_message(client[:ws], msg['message'])
+      end
+    }
   rescue => exc
     logger.error "on_rpc_message: #{exc.message}"
   end
 
   def watch_connections
-    Thread.new {
-      sleep 1 until EM.reactor_running?
```

Comment:
I wonder why this was in a separate thread with the sleep before... I assume it's perfectly fine to install a new EM timer before the EM thread is running?

Interestingly, it seems like puma actually waits for the `on_worker_boot` hook -> require `server/app/initializers/eventmachine.rb` -> thread to call `EM.run` before it will proceed with the worker boot, and thus EM will already be running when this `WebsocketBackend#initialize` -> `watch_connections` happens, even with a `sleep` to delay the `EM.run`.
atm the `WebsocketBackend#initialize` is called by puma after running the `on_worker_boot` hook which loads the initializers that start EM